readthedocs/proxito/views/serve.py

```diff
@@ -195,35 +195,40 @@ def get(self, request, proxito_path, template_name='404.html'):
         storage = get_storage_class(settings.RTD_BUILD_MEDIA_STORAGE)()
 
         # First, check for dirhtml with slash
-        for tryfile in ('index.html', 'README.html'):
-            storage_filename_path = os.path.join(
-                storage_root_path, filename, tryfile
-            )
-            log.debug(
-                'Trying index filename: project=%s version=%s, file=%s',
-                final_project.slug,
-                version_slug,
-                storage_filename_path,
-            )
-            if storage.exists(storage_filename_path):
-                log.info(
-                    'Redirecting to index file: project=%s version=%s, storage_path=%s',
+        doc_type = (
+            Version.objects.filter(project=final_project, slug=version_slug)
+            .values_list('documentation_type', flat=True)
+            .first()
+        )
+        is_htmldir_type = doc_type in (SPHINX_HTMLDIR, MKDOCS)
+        if is_htmldir_type:
+            for tryfile in ('index.html', 'README.html'):
```

Comment:
Why we don't do this in the `ServeBase.get` instead?

I mean, if we already know that this project `is_htmldir_type=True`, there is no reason to fail the first request. We can just serve the `*/index.html` in that case.

Then, we can handle _only_ the `README.html` as an special case in the 404 handler.

Reply:
We can't do that,  we serve files other than html. Like `foo/bar.img`, we'll try to serve `foo/bar.img/index.html`.

Reply:
Actually, aren't we already doing this in the `ServeBase.get` at https://github.com/readthedocs/readthedocs.org/blob/0e3df509e7810e46603be47d268273c596e68455/readthedocs/proxito/views/serve.py#L127-L132 ?

We are adding `index.html` if the URL ends with `/`, so it seems there is no need to try the same again in the 404 handler, right?

Reply:
We are only adding the index file if path ends with `/`, the 404 tries to serve the index from paths not ending with `/`